build-aux/snap/snapcraft.yaml

```diff
@@ -206,10 +207,31 @@ parts:
     override-pull: |
       craftctl default
       # set version, this needs dpkg-parsechangelog (from dpkg-dev) and git
-      craftctl set version="$(./mkversion.sh --output-only)"
+      VERSION="$(./mkversion.sh --output-only)"
+      if [ -f fips-build ] ; then
+          echo "-- appending FIPS tag to version $VERSION"
+          VERSION="$VERSION-fips"
+      fi
+      craftctl set version="$VERSION"
+
       ./get-deps.sh --skip-unused-check
     override-build: |
-      ./mkversion.sh
+      # this should be passed through build environment, but we're already
+      # anchoring dynamic linker env variables
+      GO_TOOLCHAIN_FIPS_CHANNEL="1.18-fips/stable"
+
+      VERSION="$(craftctl get version)"
+      if [ -f fips-build ] ; then
+          # use the fips channel of Go
+          snap refresh --channel "$GO_TOOLCHAIN_FIPS_CHANNEL" go
+      fi
+      # make sure to set the version we declared in pull
+      ./mkversion.sh "$VERSION"
```

Comment:
Nitpick, for sanity reasons I'd strip -fips and just set the version without calling mkversion.sh again.

Reply:
pull step called mkversion with --output-only, so no version files were generated, so we need to call it again